sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubProducerClient.cs

```diff
@@ -227,34 +224,37 @@ public EventHubProducerClient(string connectionString) : this(connectionString,
             OwnsConnection = false;
             Connection = connection;
             RetryPolicy = clientOptions.RetryOptions.ToRetryPolicy();
-            EventHubProducer = Connection.CreateTransportProducer(null, RetryPolicy);
+            PartitionProducerPool = new TransportProducerPool(Connection, RetryPolicy);
         }
 
         /// <summary>
-        ///   Initializes a new instance of the <see cref="EventHubProducerClient"/> class.
+        ///   Initializes a new instance of the <see cref="EventHubProducerClient" /> class.
         /// </summary>
         ///
         /// <param name="connection">The connection to use as the basis for delegation of client-type operations.</param>
         /// <param name="transportProducer">The transport producer instance to use as the basis for service communication.</param>
+        /// <param name="partitionProducerPool">A <see cref="TransportProducerPool" /> used to manage a set of partition specific <see cref="TransportProducer" />.</param>
         ///
         /// <remarks>
         ///   This constructor is intended to be used internally for functional
         ///   testing only.
         /// </remarks>
         ///
         internal EventHubProducerClient(EventHubConnection connection,
-                                        TransportProducer transportProducer)
+                                        TransportProducer transportProducer,
+                                        TransportProducerPool partitionProducerPool = default)
         {
             Argument.AssertNotNull(connection, nameof(connection));
             Argument.AssertNotNull(transportProducer, nameof(transportProducer));
 
             OwnsConnection = false;
             Connection = connection;
-            EventHubProducer = transportProducer;
+            RetryPolicy = new EventHubProducerClientOptions().RetryOptions.ToRetryPolicy();
```

Comment:
nit:  We can just do `new RetryOptions().ToRetryPolicy()` here.

Reply:
I made it:

```new EventHubsRetryOptions().ToRetryPolicy();```

```RetryOptions``` has an ```internal``` constructor.

Reply:
Yeah.... that's what I meant.   Sorry for the confusion and forgetting my own type name.  :smile:

_(RetryOptions is a type from Azure.Core for the HTTP pipeline)_